### ISSUE TITLE
FR: Replace "Session:" label with a session type selector (Wayland, X11 etc)

### ISSUE THREAD

**Issue by max-ishere** [rharish101/ReGreet#92]
Is this something anyone actually wants? Not theoretically, I need this theoretically-just-in-case too, practically is this a thing you want?

-----

Currently regreet has 2 options for selecting a session presented as a dropdown (X11 & Wayland combined) with a secret fallback to the login shell if there are no XDG sessions, and an arbitrary command. This works well enough, but what if the user doesn't have a default shell or they specifically want that, or what if they wanna choose a different shell from `/etc/shells`.

This is just an idea, the "Session:" label is replaced with a dropdown that has the following options (ordered by priority, default on startup is the first one):

||Source|
|---|---|
|Wayland|`$XDG_DATA_DIRS/wayland-sessions`|
|X11|`$XDG_DATA_DIRS/xsessions`|
|Default shell|`/etc/passwd`|
|Shells|`/etc/shells`|

If any of the categories were not found during loading, the option for that would be not be present.

The only issue I can anticipate is with the default shell option. If at least 1 user has it, the option would show up, but then there should be code to handle the username (regardless of it coming from the dropdown or the manual input) not having a default shell.

Handling it is not hard, just stop the user from starting auth or if the UI would in the future allow the user to change the session during auth, just put them in a state like the image below (show notification + not make it look like the auth was discarded)

![A special state showing the authentication is completed. A notification to the side of the main UI with an error about the session command being invalid](https://github.com/user-attachments/assets/e97eb689-98c6-4561-8ed3-823fc034ccb2)

At the IPC level this is the state between the last auth response (greetd returned suceess) and before start session that actually specifies the command and the ENVs.

Yes, this UI is not in this repo, its from my fork as part of #85, eventually it will be proposed as a set of PRs, I am slowly working on that.

**Comment by max-ishere** [comment]
I forgot to mention how this came into my head, I was thinking about how in #65 there is an edge case where some X11 sessions start the x server and some don't and how this could be addressed by having a separate UI for X sessions specifically, or how something like Gnome might have an X11 and Wayland modes and right now its up to the packagers of the desktop files to make sure those are actually possible to distinguish (xdg file ID being unique + including the type in the `Name` attribute.

These are not issues I have with 1 user and 1 session, so I dont know how relevant it is.

**Comment by rharish101** [comment]
IMHO, it seems like it presents too much complexity to an ordinary user, who might just want to open their DE/WM sessions. For more advanced users, they can already type in their session command.

The default shell is more of a fallback, and so I wouldn't like to expose it. As for people logging into shells, you're right, I don't think they'd be using ReGreet anyway. So I don't think that we would need to change the UI to accommodate them.

As for X11 handling, I just merged #16, which finds out if a session is an X11 session the way LightDM does. In other words, it simply checks if the parent dir is named `xsessions`. Other session desktop files should be for wayland.

**Comment by max-ishere** [comment]
So I poked in the code and it turns out the only way you'd need the default session is if either

1. The session exec key is invalid (e.g. `Exec='`
2. Nothing is selected in the combo box / text input is empty.

Both of those can be mitigated with trivial checks (I did that in my fork, but that would have to be merged after some smaller PRs). 

This is just a long way of saying the /etc/passwd shell quite useless.
